### PR TITLE
closes #2389 removed the typo in an example for derive_vars_locf_records

### LINKED ISSUE #2389: Documentation:  There is a small error in an example for function derive_locf_records
### Please select a category the issue is focused on?

Function Documentation

### Let us know where something needs a refresh or put your idea here!

Screenshot 2024-03-23 220610

In the example there is a small mistake in ADVS tribble part in second row the AVISITN is 2 but AVISIT is given as "WEEK 6"  which should be "WEEK 2"  this is relatively small but this can create some confussion in the user mind. I've highlighted the mistake part in the screenshot.

### PR DESCRIPTION
**Pull request by janardhanswami**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [ ] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Review the [Cheat Sheet](https://github.com/pharmaverse/admiral/blob/main/inst/cheatsheet/admiral_cheatsheet.pdf). Make any required updates to it by editing the file `inst/cheatsheet/admiral_cheatsheet.pptx` and re-upload a PDF version of it to the same folder.
- [ ] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Review by bms63 — COMMENTED** [review]
I am wondering if there should be a check that makes sure the AVISIT/N are mapped 1-1?  @pharmaverse/admiral WDYT?

**Comment by manciniedoardo** [comment]
> I am wondering if there should be a check that makes sure the AVISIT/N are mapped 1-1? @pharmaverse/admiral WDYT?

I think it's a good idea - where do you suggest it runs though? because if we made it its own function we'd have to add it everywhere.

**Comment by bms63** [comment]
> > I am wondering if there should be a check that makes sure the AVISIT/N are mapped 1-1? @pharmaverse/admiral WDYT?
> 
> I think it's a good idea - where do you suggest it runs though? because if we made it its own function we'd have to add it everywhere.

I think it would need to be optional??  I think we do something similar in some of the lab-focused functions with param/cd checks.  I'm not sure where it would be called...

**Comment by manciniedoardo** [comment]
> > > I am wondering if there should be a check that makes sure the AVISIT/N are mapped 1-1? @pharmaverse/admiral WDYT?
> > 
> > 
> > I think it's a good idea - where do you suggest it runs though? because if we made it its own function we'd have to add it everywhere.
> 
> I think it would need to be optional?? I think we do something similar in some of the lab-focused functions with param/cd checks. I'm not sure where it would be called...

agree, maybe it could be a function within admiraldev?

**Comment by janardhanswami** [comment]
Just wanted to know is there anything that need's to be done from my side?

**Comment by bms63** [comment]
@janardhanswami nothing needs to be done from your side.  apologies that your PR got hijacked for our discussion, but raises a good point.

The failing checks are un-related.  I will fix those for you.

**Comment by janardhanswami** [comment]
@bms63  Just wanted to know do we have any function or method so that we can check 1-1 mapping between certain variables like AVISIT and AVISITN, PARAM and PARAMCD, APERIOD and APERIODN, TRTP and TRTPN, TRTA and TRTAN. If we didn't have any function can we add one to check if the dataset  follow the 1-1 relation between certain variables  if it doesn't follow 1-1 relation then we can warn them by displaying a warning?

**Comment by bms63** [comment]
Hi @janardhanswami yes we have some of this capability built in to some functions using this - https://github.com/pharmaverse/admiraldev/blob/8a2b3d8de3d6b4b9122d4517f3a8f86bb4662ea9/R/assertions.R#L1558

I'm thinking that this function does not have this baked in.

File: R/derive_locf_records.R [hunk 1]
```diff
@@ -69,7 +69,7 @@
 #' advs <- tribble(
 #'   ~STUDYID,  ~USUBJID,      ~PARAMCD, ~PARAMN, ~AVAL, ~AVISITN, ~AVISIT,
 #'   "CDISC01", "01-701-1015", "PULSE",        1,    61,        0, "BASELINE",
-#'   "CDISC01", "01-701-1015", "PULSE",        1,    60,        2, "WEEK 6",
+#'   "CDISC01", "01-701-1015", "PULSE",        1,    60,        2, "WEEK 2",
```
Comment (bundfussr) [inline R/derive_locf_records.R:72]: I think we should change to the following to be inline with the expected visits.
```suggestion
#'   "CDISC01", "01-701-1015", "PULSE",        1,    60,        6, "WEEK 6",
```

Comment (janardhanswami) [reply R/derive_locf_records.R:72]: Hi @bundfussr do you think we should put VISIT = week 6 and AVISITN= 6  instead of AVISIT = week2 and AVISITN = 2?

Comment (bundfussr) [reply R/derive_locf_records.R:72]: Yes, because for PULSE the visits Baseline and Week 6 are expeted.
```
# A dataset with all the combinations of PARAMCD, PARAM, AVISIT, AVISITN, ... which are expected.
advs_expected_obsv <- tribble(
  ~PARAMCD, ~AVISITN, ~AVISIT,
  "PULSE",         0, "BASELINE",
  "PULSE",         6, "WEEK 6",
```

Comment (bms63) [reply R/derive_locf_records.R:72]: Hi @janardhanswami can you please update and we shall get this merged in!!  

We discussed as a team and don't think admiral should check 1-1 mappings within our functions.  It might causes too many issues with dirty data among other things.  

We are going to provide some recommendations on how to check 1-1 mappings as a pre-processing or post-processing step here: https://github.com/pharmaverse/admiral/issues/2395.  It would be great to get your review/opinion on new vignettes.  If you are interested just tag yourself in the issue so we are reminded to flag you in the review.  

Your finding generated some really great discussion.  Thanks again!

Comment (janardhanswami) [reply R/derive_locf_records.R:72]: Hi @bms63 I'll update that  and thanks for the information.

Comment (janardhanswami) [reply R/derive_locf_records.R]: Hi @bms63  I've just updated the example by removing the typo, let me know if we can merge or should I have to do anything else?

**Review by bundfussr — APPROVED** [review]

**Review by bms63 — APPROVED** [review]
@janardhanswami Thank you!

**Comment by manciniedoardo** [comment]
Hello @janardhanswami - thanks for your PR! If you like, I can include you in the admiral acknowledgements. If so can I have your full name please?

**Comment by janardhanswami** [comment]
Hi  @manciniedoardo  I would love to see my name on the admiral acknowledgements my full name is **saijanardhanswami**.
Thanks for recognizing my small efforts it boosts me a lot to even work more like this :)

**Comment by manciniedoardo** [comment]
> Hi @manciniedoardo I would love to see my name on the admiral acknowledgements my full name is **saijanardhanswami**. Thanks for recognizing my small efforts it boosts me a lot to even work more like this :) if possible please share me the link for that acknowledgement page or let me know where I can see that.

Great to hear! I will add something for Saijanard Hanswami in our June update. Please correct me if I got the spacing wrong.

**Comment by janardhanswami** [comment]
Hey @manciniedoardo  actually the spacing is wrong it should be like this  **Saijanardhan swami**  
Thanks :)

**Comment by manciniedoardo** [comment]
@janardhanswami great! thanks
